processor/src/operations/crypto_ops.rs

```diff
@@ -408,4 +448,22 @@ mod tests {
         }
         expected
     }
+
+    fn build_expected_perm(values: &[Felt]) -> [Felt; 4] {
+        let mut expected = [Felt::ZERO; 4];
+        let hash_output = hash_elements(values);
+        // reverse the result of the hash output
+        for (i, result) in hash_output.as_elements().iter().rev().enumerate() {
+            expected[i] = *result;
+        }
+        expected
+    }
```

Comment:
This function should return an array of 12 elements as we want to test the correctness of the entire hasher state after the permutation. To build the expected state we should use `apply_permutation()` function from `Rp64_256` struct.

We actually should add `apply_permutation()` pass-through function to `vm_core::hasher` module - so that we deal only with that module from here.